cmd/timescale-prometheus/main.go

```diff
@@ -329,6 +340,54 @@ func parseFlags() (*config, error) {
 	return cfg, nil
 }
 
+var errTimedOut = fmt.Errorf("timed out waiting to connect to DB")
+
+func waitForDb(cfg *config) error {
```

Comment:
Fundamentally I don't think this should return an error. If our probe failed we should still try at least once to connect to the db using the psql protocol. That way any bugs in the probe still allow the rest of the system to work.